packages/core/test/client.test.ts

```diff
@@ -597,6 +597,56 @@ describe('client', function () {
         client.notify('should not report')
       })
     })
+
+    it('modifies the notice when an async function is provided', function () {
+      const funkyName = 'My funky name'
+
+      client.beforeNotify(async (notice) => {
+        const modifyName = () => new Promise<void>((resolve) => {
+          setTimeout(() => {
+            notice.name = funkyName
+            resolve()
+          })
+        })
+
+
+        await modifyName()
+      })
+
+      return new Promise<void>((resolve) => {
+        client.afterNotify((_err, notice) => {
+          expect(notice.name).toEqual(funkyName)
+          resolve()
+        })
+
+        expect(client.notify('Should report and modify notice')).toEqual(true)
+      })
+    })
+
+    it('modifies the notice when an async function is provided', function () {
```

Comment:
Typo? This is the same test as above. Perhaps you need another test to confirm that notice won't be sent if an async handler returns false.